apps/passport-client/src/password.ts

```diff
@@ -32,18 +33,22 @@ export const setPassword = async (
   const res = await updateBlobKeyForEncryptedStorage(
     currentEncryptionKey,
     newEncryptionKey,
-    newSalt
+    newSalt,
+    knownServerStorageRevision
   );
 
-  // Meaning password is incorrect, as old row is not found
-  if (!res.success && res.error.name === "PasswordIncorrect") {
-    throw new Error(
-      "Incorrect password. If you've lost your password, reset your account below."
-    );
-  }
-
   if (!res.success) {
-    throw new Error(`Request failed with message ${res.error}`);
+    if (res.error.name === "PasswordIncorrect") {
+      throw new Error(
+        "Incorrect password. If you've lost your password, reset your account below."
+      );
+    } else if (res.error.name === "Conflict") {
+      update({ extraDownloadRequested: true });
+      throw new Error(`Cannot change password while PCDs are syncing.  
+        Wait for download to complete or reload the page and try again.`);
```

Comment:
i read the word 'conflict' here to mean that the client and server have different versions of storage, but the error message indicates a behavioral element to it, i.e. it's not just that the client and server have different storage, but also that something is in the process of happening. is 'conflict' the right name for this? are you sure that 'conflict' only results in the case that a sync is in progress?

Reply:
The message from the server means exactly what you said, and the word Conflict makes sense in the context of trying to upload something.  What I'm attempting to imply to the user is a simplification of what it means about their situation.  What's in the process of happening is that changes from some other client are in the process of propagating to this client, which I'm trying to express as "syncing is in progress".  Specifically there are some changes waiting to download.  Normally it would be waiting for the background sync to "finish syncing" (which happens on the change password screen now that I've fixed it).  By setting extraDownloadRequested to true here, I'm forcing it to happen faster, so as to make it notion of "in progress" less misleading.  If we had push/longpoll for instant updates, we could have a "sync in progress" icon which would reflect this.  In a periodic-poll world, the definition of "in progress" is a bit fuzzy.